### PR TITLE
[1.5] Fix error reporting during validation (#22281)

### PR DESCRIPTION
**Pull request by ostromart**
Manual cherry pick of https://github.com/istio/istio/pull/22281.

### CODE REVIEW

**Comment by ostromart** [comment]
@howardjohn could you please check default.yaml to see if the merged values are correct for sds and port numbers? Thanks.

File: operator/data/profiles/default.yaml [hunk 1]
```diff
@@ -648,13 +653,13 @@ spec:
         accessMode: ReadWriteMany
       zipkin:
         hub: docker.io/openzipkin
-        tag: 2.14.2
-        probeStartupDelay: 200
+        tag: 2.20.0
```
Comment (howardjohn) [inline operator/data/profiles/default.yaml]: This was not intended for 1.5 backport

Comment (ostromart) [reply operator/data/profiles/default.yaml]: do you mean just the tag or the whole block? i've changed only the tag back.

Comment (howardjohn) [reply operator/data/profiles/default.yaml]: everything (except the gateway part)

Comment (howardjohn) [reply operator/data/profiles/default.yaml]: all of this zipkin stuff though

Comment (ostromart) [reply operator/data/profiles/default.yaml]: done

**Pull request merged**